### ISSUE TITLE
Support expansion inside of address/URL bar

### ISSUE THREAD

**Issue by GaurangTandon** [GaurangTandon/ProKeys#47]
**Supports limited expansion**: any multiline text, rich text is made single line by stripping all `\n`ewlines. formatting, etc.
But, paste/date macros are completely supported. Variables are supported.
Working on the address bar while viewing a blocked site should work.

Feature is useful for people having weird proxy expansions for some reason.

**Comment by iWrote** [comment]
it's not weird, he's probably playing old games

**Comment by GaurangTandon** [comment]
This is the API:

https://developer.chrome.com/extensions/omnibox

Key takeaway:

1. we need a keyword. **How about "pk" @iWrote ?**
2. we can allow the user to type the snip text, then `searchSnippets` to search for the expected snippet, and then display it as a suggestion.
3. we can richly format the snippet suggestion.
4. when user accepts it, we can expand it inside address bar (though without rich formatting)

**Comment by GaurangTandon** [comment]
Omg how did I miss this. Yes, you are completely right. I'll try to get something ready soon.

**Comment by GaurangTandon** [comment]
Except that this behavior is exactly what happens currently. So, probably your request got obsoleted.

**Comment by ammarbinfaisal** [comment]
> this behavior is exactly what happens currently

No, this isn't the current behaviour. Currently the url stored with that snippet name gets opened. I meant there should be a way to attach something in that url. &lt;snippet-name&gt;/&lt;further-address&gt;
